### ISSUE TITLE
Issues with Rails 6.0.2.1

### ISSUE THREAD

**Issue by janosrusiczki** [pat/thinking-sphinx#1160]
I'm trying to upgrade my app from Rails 5.2.4 to Rails 6.0.2.1 and on Rails 6 anything I'm trying to do with Thinking Sphinx outputs:

    NoMethodError:  undefined method `first' for nil:NilClass
    # /home/rof/cache/bundler/ruby/2.6.0/gems/joiner-0.4.2/lib/joiner/joins.rb:22:in `alias_for'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/column_sql_presenter.rb:48:in `table'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/column_sql_presenter.rb:34:in `escape_table?'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/column_sql_presenter.rb:18:in `with_table'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/property_sql_presenter.rb:62:in `collect'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/property_sql_presenter.rb:62:in `columns_with_table'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/property_sql_presenter.rb:43:in `casted_column_with_table'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/property_sql_presenter.rb:19:in `to_select'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:70:in `collect'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:70:in `presenters_to_select'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:117:in `select_clause'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:84:in `scope_by_select'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:49:in `filter_by_scopes'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder/statement.rb:14:in `to_relation'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_builder.rb:13:in `sql_query'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_source.rb:128:in `build_sql_query'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_source.rb:153:in `prepare_for_render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/active_record/sql_source.rb:71:in `render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration/index.rb:30:in `block in render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration/index.rb:30:in `collect'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration/index.rb:30:in `render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/core/index.rb:75:in `render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration.rb:45:in `block in render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration.rb:45:in `collect'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/riddle-2.4.0/lib/riddle/configuration.rb:45:in `render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/configuration.rb:111:in `render'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/configuration.rb:119:in `block in render_to_file'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/configuration.rb:119:in `render_to_file'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/test.rb:10:in `start'
    # /home/rof/cache/bundler/ruby/2.6.0/gems/thinking-sphinx-4.4.1/lib/thinking_sphinx/test.rb:17:in `start_with_autostop'
    # ./spec/controllers/listings_controller_spec.rb:84:in `block (4 levels) in <top (required)>'

What am I doing wrong?

**Comment by pat** [comment]
Can you try running `bundle update joiner`? 

Joiner is a gem I extracted from Thinking Sphinx, and it manages the complex SQL join generation (which TS uses when creating the SQL statements in configuration files). ActiveRecord changes its internals more often than I'd like, so Joiner is updated to match specific Rails versions - but upgrading Rails itself doesn't always mean the latest Joiner release is pulled into your bundle as well. And I can confirm that Joiner v0.5.0 is the first release that supports Rails v6, but your stack-trace is using 0.4.2.

**Comment by janosrusiczki** [comment]
Yes, upgrading joiner from 0.4.2 to 0.5.0 fixed the issue. Thank you.

**Comment by boncey** [comment]
This also fixed an issue I had on Rails 5.2.
```
NoMethodError: protected method `join_root' called for #<ActiveRecord::Associations::JoinDependency:0x0000000005cb8400>
```

Fixed once I updated `joiner`.
